PWGCF/TwoParticleCorrelations/Tasks/r2p2-4-id.cxx

```diff
@@ -0,0 +1,437 @@
+  Filter globalfilter = requireGlobalTrackInFilter();
+  //---------------------------------------------------------------------------
+  HistogramRegistry histos{"R2P2", {}, OutputObjHandlingPolicy::AnalysisObject};
+
+  void init(InitContext const&)
+  {
+    CommonInit2(histos);
+  }
+  void process(soa::Filtered<soa::Join<aod::Collisions, aod::EvSels>>::iterator const& filteredCollision, soa::Filtered<soa::Join<aod::Tracks, aod::TrackSelection, aod::pidTPCPi, aod::pidTOFPi, aod::pidTPCPr, aod::pidTOFPr, aod::pidTPCKa, aod::pidTOFKa, aod::pidTPCEl, aod::pidTOFbeta, aod::TracksExtra>> const& tracks)
+  {
+    CommonProcess2(tracks, histos, 1, 2);
+  }
+};
+
+// Pion-Proton pair correlation
+struct r2p24pip {
+  //-----Track&Event Selection-------------------------------------------------
+  Filter col = aod::evsel::sel8 == true;
+  Filter collisionFilter = (nabs(aod::collision::posZ) < 10.f);
+  Filter ptfilter = aod::track::pt > 0.2f && aod::track::pt < 2.0f;
+  Filter globalfilter = requireGlobalTrackInFilter();
+  //---------------------------------------------------------------------------
+  HistogramRegistry histos{"R2P2", {}, OutputObjHandlingPolicy::AnalysisObject};
+
+  void init(InitContext const&)
+  {
+    CommonInit2(histos);
+  }
+  void process(soa::Filtered<soa::Join<aod::Collisions, aod::EvSels>>::iterator const& filteredCollision, soa::Filtered<soa::Join<aod::Tracks, aod::TrackSelection, aod::pidTPCPi, aod::pidTOFPi, aod::pidTPCPr, aod::pidTOFPr, aod::pidTPCKa, aod::pidTOFKa, aod::pidTPCEl, aod::pidTOFbeta, aod::TracksExtra>> const& tracks)
+  {
+    CommonProcess2(tracks, histos, 1, 3);
+  }
+};
+
+// Proton-Kaon pair correlation
+struct r2p24pk {
+  //-----Track&Event Selection-------------------------------------------------
+  Filter col = aod::evsel::sel8 == true;
+  Filter collisionFilter = (nabs(aod::collision::posZ) < 10.f);
+  Filter ptfilter = aod::track::pt > 0.2f && aod::track::pt < 2.0f;
+  Filter globalfilter = requireGlobalTrackInFilter();
+  //---------------------------------------------------------------------------
+  HistogramRegistry histos{"R2P2", {}, OutputObjHandlingPolicy::AnalysisObject};
+
+  void init(InitContext const&)
+  {
+    CommonInit2(histos);
+  }
+  void process(soa::Filtered<soa::Join<aod::Collisions, aod::EvSels>>::iterator const& filteredCollision, soa::Filtered<soa::Join<aod::Tracks, aod::TrackSelection, aod::pidTPCPi, aod::pidTOFPi, aod::pidTPCPr, aod::pidTOFPr, aod::pidTPCKa, aod::pidTOFKa, aod::pidTPCEl, aod::pidTOFbeta, aod::TracksExtra>> const& tracks)
+  {
+    CommonProcess2(tracks, histos, 3, 2);
+  }
+};
+
+WorkflowSpec defineDataProcessing(ConfigContext const& cfgc)
+{
+  return WorkflowSpec{
+    adaptAnalysisTask<r2p24ch>(cfgc),
+    adaptAnalysisTask<r2p24pi>(cfgc),
+    adaptAnalysisTask<r2p24ka>(cfgc),
+    adaptAnalysisTask<r2p24pr>(cfgc),
+    adaptAnalysisTask<r2p24pik>(cfgc),
+    adaptAnalysisTask<r2p24pip>(cfgc),
+    adaptAnalysisTask<r2p24pk>(cfgc),
+  };
+}
```

Comment:
Please, consider if with my previous comments this structure can be simplified 

With the current structure the processing per collision (basically going trough the double inner tracks loop) is repeated for each of the tasks that, in principle as they are defined, will be always executed in parallel. For the CPU consuming point of view this will definitely be penalized 

Consider the tagging of the tracks as described, which will need the tagging task, and then a single task which should incorporate the double inner tracks loop which, with the proper array of histograms, should fill all the needed histograms requiring only one double inner tracks loop